examples/list_pod/main.c

```diff
@@ -0,0 +1,111 @@
+#include <apiClient.h>
+#include <CoreV1API.h>
+#include <malloc.h>
+#include <stdio.h>
+#include <errno.h>
+
+#define K8S_APISERVER_BASEPATH "https://9.111.254.254:6443"
+#define K8S_TOKEN_FILE_IN_CLUSTER "/var/run/secrets/kubernetes.io/serviceaccount/token"
+#define K8S_TOKEN_BUF_SIZE 1024
+#define K8S_AUTH_KEY "Authorization"
+#define K8S_AUTH_VALUE_TEMPLATE "Bearer %s"
+
+apiClient_t *g_k8sAPIConnector;
+
+void list_pod(apiClient_t *apiClient)
+{
+    v1_pod_list_t *pod_list = NULL;
+    pod_list = CoreV1API_listNamespacedPod( apiClient ,
+                                            "default", /*namespace*/
+                                            NULL, /* pretty */
+                                            0, /* allowWatchBookmarks */
+                                            NULL, /* continue */
+                                            NULL, /* fieldSelector */
+                                            NULL, /* labelSelector */
+                                            0, /* limit */
+                                            NULL, /* resourceVersion */
+                                            0, /* timeoutSeconds */
+                                            0 /* watch */
+                    );
+    printf("return code=%ld\n", apiClient->response_code);
+    if(pod_list){
+        printf("Get pod list.\n");
+        listEntry_t *listEntry = NULL;
+        v1_pod_t *pod = NULL;
+        list_ForEach(listEntry, pod_list->items){
+            pod = listEntry->data;
+            printf("pod name=%s\n", pod->metadata->name);
+        }
+
+    }else{
+        printf("Cannot list any pod.\n");
+    }
+}
+
+int
+loadK8sConfigInCluster(char *token, int token_buf_size)
+{
+    static char fname[] = "loadK8sConfigInCluster()";
+
+    FILE *fp;
+    fp = fopen(K8S_TOKEN_FILE_IN_CLUSTER, "r");
+
+    if (fp == NULL) {
+        if (errno == ENOENT) {
+            printf("\
+%s: The file %s does not exist.",
```

Comment:
why is there a line-break here?

Reply:
No special reason, just want to keep the message text in a dedicated line.

Removing the line-break is OK to me.